modulus/datapipes/geometric/vtk_datapipe.py

```diff
@@ -0,0 +1,465 @@
+
+            # All workers use the same rng seed so the resulting
+            # indices are the same across workers.
+            np.random.default_rng(seed=sample_info.epoch_idx).shuffle(self.indices)
+            self.last_epoch = sample_info.epoch_idx
+
+        # Get local indices from global index.
+        idx = self.indices[sample_info.idx_in_epoch]
+
+        # if self.poly_data is None:  # TODO check
+        # This will be called once per worker. Workers are persistent,
+        # so there is no need to explicitly close the files - this will be done
+        # when corresponding pipeline/dataset is destroyed.
+        data = self.vtk_reader_fn(self.data_paths[idx])
+        # vertices, pressure, wss = self.parse_vtkpolydata(polydata)
+
+        return self.parse_vtk_data_fn(data, self.vars)
+
+    def __len__(self):
+        return len(self.indices)
+
+    def vtk_reader(self):
+        if self.file_format == "vtp":
+            return read_vtp
+        elif self.file_format == "vtu":
+            return read_vtu
+        elif self.file_format == "cgns":
+            return read_cgns
+        else:
+            raise NotImplementedError(
+                f"Data type {self.file_format} is not supported yet"
+            )
+
+    def parse_vtk_data(self):
+        if self.file_format == "vtp":
+            return _parse_vtk_polydata
+        elif self.file_format == "vtu":
+            return _parse_vtk_unstructuredgrid
+        else:
+            raise NotImplementedError(
+                f"Data type {self.file_format} is not supported yet"
+            )
+
+
+def _parse_vtk_polydata(polydata, vars):
+    # Fetch vertices
+    points = polydata.GetPoints()
+    if points is None:
+        raise ValueError("Failed to get points from the polydata.")
+    vertices = torch.tensor(
+        np.array([points.GetPoint(i) for i in range(points.GetNumberOfPoints())]),
+        dtype=torch.float32,
+    )
+
+    # Fetch node attributes  # TODO modularize
+    attributes = []
+    point_data = polydata.GetPointData()
+    if point_data is None:
+        raise ValueError("Failed to get point data from the unstructured grid.")
+    for array_name in vars:
+        try:
+            array = point_data.GetArray(array_name)
+        except ValueError:
+            raise ValueError(
+                f"Failed to get array {array_name} from the unstructured grid."
+            )
+        array_data = np.zeros(
+            (points.GetNumberOfPoints(), array.GetNumberOfComponents())
+        )
+        for j in range(points.GetNumberOfPoints()):
+            array.GetTuple(j, array_data[j])
+        attributes.append(torch.tensor(array_data, dtype=torch.float32))
+    attributes = torch.cat(attributes, dim=-1)
+
+    # Fetch edges
+    polys = polydata.GetPolys()
+    if polys is None:
+        raise ValueError("Failed to get polygons from the polydata.")
+    polys.InitTraversal()
+    edges = []
+    for i in range(polys.GetNumberOfCells()):
```

Comment:
Same comment as here: https://github.com/NVIDIA/modulus/pull/544#discussion_r1635177257

Reply:
I'm not sure about this. Maybe we should keep an eye on this. I checked the number of cells and it exactly matches with the input data.

Reply:
I don't think the cells is the issue here, but rather the edges. Again, might be ok for training a GNN, but if we don't capture all the edges, that sounds a bit problematic to me.

Reply:
I fixed this according to your suggested fix for the stokes flow example.

Reply:
Great, thank you!